KIPs/kip-71.md

```diff
@@ -0,0 +1,381 @@
+# In this proposal, those concatenation is intentionally ommited for simplification.
+
+# Transaction2930 in Ethereum. 
+@dataclass
+class TxTypeEthereumAccessList:
+	type: int = 0x7801
+	chain_id: int = 0
+	nonce: int = 0
+	gas_price: int = 0
+	gas: int = 0
+	to: int = 0
+	value: int = 0
+	data: bytes = bytes()
+	access_list: List[Tuple[int, List[int]]] = field(default_factory=list)
+	v: int = 0
+	r: int = 0
+	s: int = 0
+
+# Transaction1559 in Ethereum.
+@dataclass
+class TxTypeEthereumDynamicFee:
+	type: int = 0x7802
+	chain_id: int = 0
+	nonce: int = 0
+	gas_tip_cap: int = 0
+	gas_fee_cap: int = 0
+	gas: int = 0
+	to: int = 0
+	value: int = 0
+	data: bytes = bytes()
+	access_list: List[Tuple[int, List[int]]] = field(default_factory=list)
+	v: int = 0
+	r: int = 0
+	s: int = 0
+
+
+EthereumTransactions = Union[TxTypeEthereumDynamicFee, TxTypeEthereumAccessList]
+
+Transaction = Union[TxTypeLegacyTransaction, TxTypeFeeDelegatedSmartContractExecution, TxTypeFeeDelegatedSmartContractExecutionWithRatio, EthereumTransactions]
+
+# TODO: transaction accounting part
+@dataclass
+class NormalizedTransaction:
+	signer_address: int = 0
+	signer_nonce: int = 0
+	max_fee_per_gas: int = 0
+	gas_limit: int = 0
+	to: int = 0
+	value: int = 0
+	data: bytes = bytes()
+	fee_payer_address: int = 0
+	fee_ratio: int = 1
+
+@dataclass
+class Block:
+	hash: int = 0
+	parent_hash: int = 0
+	base_fee_per_gas: int = 0
+	block_score: int = 0
+	extra_data: bytes = bytes()
+	gas_used: int = 0 
+	governance_data: bytes = bytes()
+	logs_bloom: int = 0
+	number: int = 0
+	transaction_receipt_root: int = 0
+	reward: int = 0
+	state_root: int = 0
+	timestamp: int = 0
+	timestamp_FoS: int = 0
+	transaction_root: int = 0
+	nonce: int = 0
+	committee: List[int] = field(default_factory=list)
+	proposer: int = 0 
+	size: int = 0 
+
+
+@dataclass
+class Account:
+	type: int = 0
+	nonce: int = 0
+	humanReadable: bool = False 
+	address: int = 0
+	key: int = 0 
+	balance: int = 0
+	storage_root: int = 0
+	code_hash: int = 0
+	code_format: int = 0
+	vm_version: int = 0
+
+INITIAL_FORK_BLOCK_NUMBER = 107806544 # TBD
+BASE_FEE_DELTA_REDUCING_DENOMINATOR = 64 # TBD
+LOWER_BOUND_BASE_FEE = 2000000000 # TBD, 20ston
+UPPER_BOUND_BASE_FEE = 2000000000000 # TBD, 2000ston
+GAS_TARGET = 30000000 # TBD
+BURN_RATIO = 0.5 # TBD
+
+class World(ABC):
+	def validate_block(self, block: Block) -> None:
+		
+		# check if the base fee is correct
+		if INITIAL_FORK_BLOCK_NUMBER == block.number:
+			expected_base_fee_per_gas = LOWER_BOUND_BASE_FEE
+
+		else: 
+			parent_base_fee_per_gas = self.parent(block).base_fee_per_gas
+			parent_gas_used = self.parent(block).gas_used
+			transactions = self.transactions(block)
+
+			# check if the base fee is in the range
+			if parent_base_fee_per_gas < LOWER_BOUND_BASE_FEE:
+				expected_base_fee_per_gas = LOWER_BOUND_BASE_FEE
+
+			elif parent_base_fee_per_gas > UPPER_BOUND_BASE_FEE:
+				expected_base_fee_per_gas = UPPER_BOUND_BASE_FEE
+
```

Comment:
Don't we need to first calculate the expected_base_fee_per_gas?
then if the value is not in the range (lower, upper), we need to adjust the value.